runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -780,23 +780,91 @@ J9::CodeGenerator::lowerTreeIfNeeded(
+         if (is64Bit)
             {
-            numArgsNode = TR::Node::lconst(node, numParameterStackSlots);
-            storeNode = TR::Node::createStore(vmThreadTempSlotSymRef, numArgsNode, TR::lstore);
+            storeOpCode = TR::lstore;
+            numArgSlotsNode = TR::Node::lconst(node, numParameterStackSlots);
             }
          else
             {
-            numArgsNode = TR::Node::iconst(node, numParameterStackSlots);
-            storeNode = TR::Node::createStore(vmThreadTempSlotSymRef, numArgsNode, TR::istore);
+            storeOpCode = TR::istore;
+            numArgSlotsNode = TR::Node::iconst(node, numParameterStackSlots);
+            }
+         tempSlotStoreNode = TR::Node::createStore(self()->comp()->getSymRefTab()->findOrCreateVMThreadTempSlotFieldSymbolRef(),
+                                                   numArgSlotsNode,
+                                                   storeOpCode);
+         tempSlotStoreNode->setByteCodeIndex(node->getByteCodeIndex());
+         TR::TreeTop::create(self()->comp(), tt->getPrevTreeTop(), tempSlotStoreNode);
+
+         if (rm == TR::java_lang_invoke_MethodHandle_linkToStatic || rm ==  TR::java_lang_invoke_MethodHandle_linkToSpecial)
+            {
+            int32_t numArgs;
+            if (node->getSymbolReference()->getSymbol()->isDummyResolvedMethod())
+               {
+               // The last child of the call node is the iload for the memberName object, from which we can obtain the callSite index or cpIndex.
+               TR::Node * invokeCacheArrayShadowNode = node->getLastChild();
+
+               // In compressed pointers mode, the compressed refs anchors are transformed right before codegen, for which we
+               // may need to go a few levels down to reach the invoke cache array shadow node
+               //
+               // n28n      compressedRefs
+               // n26n        l2a
+               // n49n          lshl (compressionSequence )
+               // n48n            iu2l
+               // n47n              iloadi  <array-shadow>[#229  Shadow] [flags 0x80000607 0x0 ]
+               // n25n                aladd (X>=0 internalPtr sharedMemory )
+               // n5n                   ==>aload
+               // n24n                  lconst 16 (highWordZero X!=0 X>=0 )
+               // n46n            iconst 3
+               // n15n        ==>lconst 0
+               //
+               if (self()->comp()->useCompressedPointers())
+                  {
+                  TR::Node * currentNode = invokeCacheArrayShadowNode;
+                  while (currentNode){
+                     if (currentNode->getOpCode().isLoad()) break;
+                     currentNode = currentNode->getFirstChild();
+                  }
+                  invokeCacheArrayShadowNode = currentNode;
+                  }
+
+               TR::Node * sideTableEntryNode = invokeCacheArrayShadowNode->getFirstChild()->getFirstChild();
```

Comment:
I'm always nervous to see multiple reaches like this.  Can you add a fatal assert (with node) before this such that `invokeCacheArrayShadowNode->getFirstChild()` is not null?